application-collabora-default/src/main/java/com/xwiki/collabora/internal/FileToken.java

```diff
@@ -154,6 +156,30 @@ public void setHasEdit(boolean hasEdit)
         this.hasEdit = hasEdit;
     }
 
+    /**
+     * @return the last requested action mode. This may be either "edit" or "view".
+     */
+    public String getLastActionMode()
+    {
+        return lastActionMode;
+    }
+
+    /**
+     * @param lastActionMode the last requested action mode. This may be either "edit" or "view".
+     */
+    public void setLastActionMode(String lastActionMode)
+    {
+        this.lastActionMode = lastActionMode;
+    }
+
+    /**
+     * @return {@code true} if the last requested action was edit, or {@code false} otherwise.
+     */
+    public boolean requestedEdit()
```

Comment:
I think this method gives enough information so you won't need to use the `get` one. Maybe you could only keep this one?

```suggestion
    public boolean isEditRequested()
```